### ISSUE TITLE
SSL / Proxy requests

### ISSUE THREAD

**Issue by jroakes** [JakobGreen/lua-requests#6]
Is there anyway to make get requests via this codebase using a proxy for https urls?  Currently getting "proxy not supported"

**Comment by jroakes** [comment]
Changing  line 102  to ` local socket = string.find(full_request.url, '^https:') and not request.proxy and https_socket or http_socket` sent the proxy through socket.http and fixed the issue.

**Comment by JakobGreen** [comment]
Good catch. I made the [fix](https://github.com/JakobGreen/lua-requests/commit/c78e0ca25a20348dfbdd1660d5dadac46061bfd3) and uploaded a new luarock.

**Comment by jroakes** [comment]
hmm.  even with that, I noticed that I am getting <head><title>400 The plain HTTP request was sent to HTTPS port</title></head> using a proxy with the http transport on nginx sites (the ones I need).  Any ideas.  This is the closest thing I have found to working in lua for now.  It needs a solid http/https library badly.

**Comment by JakobGreen** [comment]
Yeah unfortunately it looks like this isn't possible using LuaSec and LuaSocket how they currently are. Have you tried using the [lua curl bindings](https://github.com/Lua-cURL/Lua-cURLv3)? I know cURL supports different proxies pretty well so that might be able to do it. I have been considering trying it out for this library instead of LuaSec and LuaSocket because cURL has so many more features.
